### PR TITLE
build: upgrade to Go 1.17

### PR DESCRIPTION
**Pull request by Juneezee**
This pull request introduces three changes:

1. Enable  [module graph pruning](https://golang.org/ref/mod#graph-pruning) and [lazy module loading](https://golang.org/ref/mod#lazy-loading) by upgrading the `go` directive in `go.mod` to `1.17`. 

1. Upgrade the Go image in Dockerfile to `1.17-alpine`.

1. Replaces the existing `io/ioutil` functions with their new definitions in `os` packages in `main.go`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -66,7 +66,7 @@ RUN --mount=target=.,from=src,src=/src/qemu,rw --mount=target=./install-scripts,
 ARG BINARY_PREFIX
 RUN cd /usr/bin; [ -z "$BINARY_PREFIX" ] || for f in $(ls qemu-*); do ln -s $f $BINARY_PREFIX$f; done
 
-FROM --platform=$BUILDPLATFORM golang:1.16-alpine AS binfmt
+FROM --platform=$BUILDPLATFORM golang:1.17-alpine AS binfmt
```
Comment (tonistiigi) [inline Dockerfile:69]: Could you move this into build arg like in https://github.com/docker/buildx/blob/master/Dockerfile#L3 so we only need to update one place in the future.

**Comment by tonistiigi** [comment]
Check out the CI errors as well.

**Comment by Juneezee** [comment]
@tonistiigi The CI errors are fixed :smiley: Please take a look again. Thank you.

**Review by tonistiigi — APPROVED** [review]

**Comment by tonistiigi** [comment]
Thanks

**Pull request merged**